internal/service/assessment.go

```diff
@@ -91,6 +95,109 @@ func (as *AssessmentService) GetAssessment(ctx context.Context, id uuid.UUID) (*
 	return assessment, nil
 }
 
+// CreateRvtoolsAssessment creates an assessment with RVTools processing
+// It creates an empty assessment and snapshot, enqueues a River job, and returns immediately
+func (as *AssessmentService) CreateRvtoolsAssessment(ctx context.Context, createForm mappers.AssessmentCreateForm) (*model.Assessment, error) {
+	logger := as.logger.WithContext(ctx)
+	tracer := logger.Operation("create_rvtools_assessment").
+		WithString("org_id", createForm.OrgID).
+		WithString("name", createForm.Name).
+		Build()
+
+	assessment := createForm.ToModel()
+	tracer.Step("convert_form_to_model").WithUUID("assessment_id", assessment.ID).Log()
+
+	// Validate file data
+	if len(createForm.RVToolsFile) == 0 {
+		return nil, fmt.Errorf("rvtools file is empty")
+	}
+	tracer.Step("validate_file").WithInt("file_size", len(createForm.RVToolsFile)).Log()
+
+	// Encode to base64 for storage in job args
+	encodedData := base64.StdEncoding.EncodeToString(createForm.RVToolsFile)
+	tracer.Step("encoded_base64").WithInt("encoded_size", len(encodedData)).Log()
+
+	// Start transaction
+	var err error
+	ctx, err = as.store.NewTransactionContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_, _ = store.Rollback(ctx)
+	}()
+
+	// Create pending snapshot for async processing
+	createdSnapshot := model.Snapshot{
+		Status: model.SnapshotStatusPending,
+	}
+
+	// Create assessment with snapshot
+	if err := as.store.Assessment().Create(ctx, &assessment, &createdSnapshot); err != nil {
+		_, _ = store.Rollback(ctx)
+		if errors.Is(err, store.ErrDuplicateKey) {
+			return nil, NewErrAssessmentDuplicateName(assessment.Name)
+		}
+		return nil, fmt.Errorf("failed to create assessment: %w", err)
+	}
+
+	tracer.Step("assessment_and_snapshot_created").
+		WithUUID("created_assessment_id", assessment.ID).
+		WithInt("snapshot_id", int(createdSnapshot.ID)).
+		WithString("snapshot_status", string(createdSnapshot.Status)).
+		Log()
+
+	// Commit transaction before enqueuing job
+	ctx, err = store.Commit(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Enqueue River job for async processing
+	if as.riverClient != nil {
```

Comment:
what are the scenarios when the client could be nil? This is a existential condition here. no riverClient no rvtool processing